tests/integration/test_charm.py

```diff
@@ -5,19 +5,104 @@
 import logging
 
 import pytest
+from helpers import (
+    check_prometheus_is_ready,
+    get_job_config_for,
+    get_prometheus_config,
+    get_prometheus_rules,
+    get_rules_for,
+)
 
 logger = logging.getLogger(__name__)
 
 
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy_with_ubuntu_image(ops_test):
+@pytest.mark.parametrize("oci_image", ["ubuntu/prometheus:latest", "prom/prometheus:latest"])
```

Comment:
That is @simskij call. I am not aware if we are version fixing images for our charms.

Reply:
The Kubeflow team tend to use the `upstream-image` parameter in their `metadata.yaml` pretty much specifically for enabling this kind of testing. We also don't really need to support both images, picking one (probably the `ubuntu` one) is fine.

Reply:
Thought I posted a comment on this yesterday. I'll try to recollect what I wrote then.

The charm locks down the digest for the image on publish (see for instance https://charmhub.io/prometheus-k8s/resources/prometheus-image). 

I agree that we should pin it down to whatever is available in the published charm(s) for charms **other** than the one we're testing **from**. For instance, if we were to do an integration test with prometheus and grafana, from this repo, I'd strongly suggest we use the digest that is published for the grafana charm rather than `latest`.

For the charm under test on the other hand, I think it makes sense to use whatever digest would be set in the next revision if we were to publish the code we're testing, rather than what is already there.

Reply:
I did not use `upstream-image` as I was asked to remove this in a previous code review. So should I remove the use upstream Prometheus image and add back `upstream-image` in `metadata.yaml` ?

Reply:
I recall @jnsgruk had agreed using the upstream-image tag was useful both as docu for the end user, and for these testing scenarios

Reply:
If we're going with only one image then yes, using upstream image is easiest. I use it in alertmanager exactly for this.

Reply:
So only test with Ubuntu latest ?

Reply:
There has been a lot of back-and-forth in PM land about what we can support, and it is not settled yet. As of now, the ROCK is the only thing we can _certainly_ support so let’s go with that.

Reply:
OK. Removed test with upstream Prometheus image retaining it only with ubuntu latest rock.